src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/scaled_dot_product_attention_decomposition.hpp"
+
+#include <memory>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/gather.hpp"
+#include "openvino/op/greater_eq.hpp"
+#include "openvino/op/logical_not.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/range.hpp"
+#include "openvino/op/scaled_dot_product_attention.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/softmax.hpp"
+#include "openvino/op/sqrt.hpp"
+#include "openvino/op/squeeze.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+ov::pass::ScaledDotProductAttentionDecomposition::ScaledDotProductAttentionDecomposition() {
+    MATCHER_SCOPE(ScaledDotProductAttentionDecomposition);
+    auto pattern_node = ov::pass::pattern::wrap_type<ov::op::v13::ScaledDotProductAttention>();
+
+    matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        auto& pattern_to_output = m.get_pattern_value_map();
+        auto node = std::dynamic_pointer_cast<ov::op::v13::ScaledDotProductAttention>(
+            pattern_to_output.at(pattern_node).get_node_shared_ptr());
+
+        if (node == nullptr || transformation_callback(node)) {
+            return false;
+        }
+
+        auto new_output_node = decompose(node)[0].get_node_shared_ptr();
+        new_output_node->set_friendly_name(node->get_friendly_name());
+        ov::copy_runtime_info(node, new_output_node);
```

Comment:
Usually the runtime info is copied to all of the new nodes created during the transformation by collecting them in a vector or by usage of `NodeRegistry`.
See unique_decomposition as example:
https://github.com/openvinotoolkit/openvino/blob/475ddb9283abd8040f0104572c0c1ace733fb126/src/common/transformations/src/transformations/op_conversions/unique_decomposition.cpp#L58 https://github.com/openvinotoolkit/openvino/blob/475ddb9283abd8040f0104572c0c1ace733fb126/src/common/transformations/src/transformations/op_conversions/unique_decomposition.cpp#L81-L82 https://github.com/openvinotoolkit/openvino/blob/475ddb9283abd8040f0104572c0c1ace733fb126/src/common/transformations/src/transformations/op_conversions/unique_decomposition.cpp#L202

Reply:
@itikhono, @jane-intel, do we have a simpler way to copy_runtime_info to entire subgraph based on topological properties without collecting all nodes in decomposition? There are many nodes, and it becomes bulky.

Reply:
`MatcherPass::register_new_node` made the trick, no need to use `NodeRegistry` directly.